tests/model.rs

```diff
@@ -0,0 +1,35 @@
+extern crate micrograd_rs;
+use indexmap::IndexMap;
+use micrograd_rs::prelude::*;
+use micrograd_rs::{Activation, Layer, Linear, Model, Sequential};
+
+#[test]
+fn valid_load_state_dict() {
+    let mut model = sequential!(
+        Ix1,
+        [
+            Linear::new(String::from("fc1"), 4, 3),
+            Activation::ReLu,
+            Linear::new(String::from("fc2"), 3, 2),
+            Activation::ReLu
+        ]
+    );
+
+    let mut state_dict = IndexMap::new();
+    state_dict.insert(
+        String::from("fc1.weight"),
+        vec![
+            0.1, -0.2, 0.3, -0.4, 0.5, -0.6, 0.7, -0.8, 0.9, -0.1, 0.2, -0.3,
+        ],
+    );
+    state_dict.insert(String::from("fc1.bias"), vec![0.1, -0.2, 0.3]);
+    state_dict.insert(
+        String::from("fc2.weight"),
+        vec![-0.1, 0.2, -0.3, -0.4, 0.5, -0.6],
+    );
+    state_dict.insert(String::from("fc2.bias"), vec![-0.4, 0.5]);
+
+    model.load_state_dict(&state_dict);
+
+    assert_eq!(model.state_dict(), state_dict);
```

Comment:
Might be nice to include a `assert_ne!` before loading the state dict to show that it actually was different